raft/src/test/java/org/apache/kafka/snapshot/SnapshotWriterReaderTest.java

```diff
@@ -135,4 +141,18 @@ public static void assertSnapshot(List<List<String>> batches, SnapshotReader<Str
 
         assertEquals(expected, actual);
     }
+
+    private RaftClientTestContext initContextAsLeaderAndAdvanceHighWatermark(OffsetAndEpoch snapshotId) throws Exception {
```

Comment:
Okay but note that if the quorum is a single node quorum then the high-watermark should advance simply by `context.client.poll()`. You can simplify this if you want by making the quorum a single member quorum. See the test in `KafkaRaftClientTest.testLeaderAppendSingleMemberQuorum()`.

Reply:
Cool, updated :)